src/o365/pa/commands/pcf/pcf-init.ts

```diff
@@ -0,0 +1,206 @@
+        return 'The current directory cannot be used because it already contains a project. Please create a new directory and retry the operation.';
+      }
+
+      const workingDirectoryName: string = path.basename(process.cwd());
+      if (!workingDirectoryName 
+        || /^((\..*)|COM\d|CLOCK\$|LPT\d|AUX|NUL|CON|PRN|(.*[\u{d800}-\u{dfff}]+.*))$/iu.test(workingDirectoryName)
+        || /^(.*\.\..*)$/i.test(workingDirectoryName))
+      {
+          return `Empty or invalid project name '${workingDirectoryName}'`;
+      }
+
+      if (args.options.name){
+        if (!/^(?!\d)[a-zA-Z0-9]+$/i.test(args.options.name)) {
+          return `Value of 'name' is invalid. Only characters within the ranges [A - Z], [a - z] or [0 - 9] are allowed. The first character may not be a number.`;
+        }
+
+        if (Utils.isJavascriptReservedWord(args.options.name)) {
+          return `The value '${args.options.name}' passed for 'name' is a reserved word.`;
+        }
+      }
+      else {
+        return 'Missing required option name.';
+      }
+
+      if (args.options.namespace) {
+        if (!/^(?!\.|\d)(?!.*\.$)(?!.*?\.\d)(?!.*?\.\.)[a-zA-Z0-9.]+$/i.test(args.options.namespace)) {
+          return `Value of 'namespace' is invalid. Only characters within the ranges [A - Z], [a - z], [0 - 9], or '.' are allowed. The first and last character may not be the '.' character. Consecutive '.' characters are not allowed. Numbers are not allowed as the first character or immediately after a period.`;
+        }
+
+        if (Utils.isJavascriptReservedWord(args.options.namespace)) {
+          return `The value '${args.options.namespace}' passed for 'namespace' is or includes a reserved word.`;
+        }
+      } 
+      else {
+        return 'Missing required option namespace.';
+      }
+
+      if (args.options.namespace && args.options.name && (args.options.namespace + args.options.name).length > 75)
+      {
+        return `The total length of values for 'name' and 'namespace' cannot exceed 75. Length of 'name' is ${args.options.name.length}, length of 'namespace' is ${args.options.namespace.length}.`;
+      }
+
+      if (args.options.template) {
+        const testTemplate: string = args.options.template.toLowerCase();
+        if (!(testTemplate === 'field' || testTemplate === 'dataset')) {
+          return `Template must be either 'Field' or 'Dataset', but '${args.options.template}' was provided.`;
+        }
+      }
+      else {
+        return 'Missing required option template.';
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: CommandArgs, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(commands.PCF_INIT).helpInformation());
+    log(
+      `  Remarks:
+
+    Name cannot contain reserved Javascript words. Only characters within the ranges [A - Z], [a - z] or [0 - 9] are allowed. The first character may not be a number.
```

Comment:
We should wrap these longer lines because otherwise they will be hard to read in terminal.